js/src/ui/Form/TypedInput/typedInput.js

```diff
@@ -61,8 +67,11 @@ export default class TypedInput extends Component {
   };
 
   componentWillMount () {
-    if (this.props.isEth && this.props.value) {
-      this.setState({ isEth: true, ethValue: fromWei(this.props.value) });
+    const { isEth, value } = this.props;
+
+    if (typeof isEth === 'boolean' && value) {
```

Comment:
So the logic is :

- If `isEth` is a prop (`false` or `true`), then the `ETH` toggle will be shown for `integer` inputs
The prop value will then be the default toggle state (ie.`false` doesn't convert the value to `ETH` by default)
- Else (ie. `undefined`, not a `boolean`), don't show the toggle at all